app/scripts/modules/azure/loadBalancer/configure/AzureLoadBalancerChoiceModal.tsx

```diff
@@ -0,0 +1,101 @@
+import * as React from 'react';
+import { Button, Modal } from 'react-bootstrap';
+
+import { ILoadBalancerModalProps, ModalClose, ReactModal, noop, CloudProviderRegistry } from '@spinnaker/core';
+
+import { AzureLoadBalancerTypes, IAzureLoadBalancer } from '../../utility';
+import { ModalInjector } from 'core/reactShims';
+
+export interface IAzureLoadBalancerChoiceModalState {
+  choices: IAzureLoadBalancer[];
+  selectedChoice: IAzureLoadBalancer;
+}
+
+export class AzureLoadBalancerChoiceModal extends React.Component<
+  ILoadBalancerModalProps,
+  IAzureLoadBalancerChoiceModalState
+> {
+  private static parentProps: ILoadBalancerModalProps;
+  public static defaultProps: Partial<ILoadBalancerModalProps> = {
+    closeModal: noop,
+    dismissModal: noop,
+  };
+
+  public static show(props: ILoadBalancerModalProps): Promise<void> {
+    this.parentProps = props;
```

Comment:
What is the purpose of `parentProps`?

Reply:
Removed it and use this.props.app to inject application property